persistence-jdbc/scaladsl/src/main/scala/com/lightbend/lagom/scaladsl/persistence/jdbc/JdbcPersistenceComponents.scala

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package com.lightbend.lagom.scaladsl.persistence.jdbc
+
+import scala.concurrent.ExecutionContext
+
+import akka.actor.ActorSystem
+import com.lightbend.lagom.internal.persistence.jdbc.SlickProvider
+import com.lightbend.lagom.internal.scaladsl.persistence.ReadSideConfig
+import com.lightbend.lagom.internal.scaladsl.persistence.jdbc.JdbcOffsetStore
+import com.lightbend.lagom.internal.scaladsl.persistence.jdbc.JdbcReadSideImpl
+import com.lightbend.lagom.internal.scaladsl.persistence.jdbc.OffsetTableConfiguration
+import play.api.Configuration
+import play.api.db.DBApi
+import com.lightbend.lagom.internal.scaladsl.persistence.jdbc.JdbcPersistentEntityRegistry
+import com.lightbend.lagom.scaladsl.persistence.PersistentEntityRegistry
+import com.lightbend.lagom.internal.scaladsl.persistence.jdbc.JdbcSessionImpl
+import com.lightbend.lagom.internal.scaladsl.persistence.OffsetStore
+
+/**
+ * Persistence JDBC components (for compile-time injection).
+ */
+trait JdbcPersistenceComponents {
+
+  def system: ActorSystem
+  def executionContext: ExecutionContext
+  def configuration: Configuration
+  def dbApi: DBApi // FIXME where is this created?
+  def readSideConfig: ReadSideConfig
+
+  lazy val slickProvider: SlickProvider = new SlickProvider(system, dbApi)(executionContext)
+  lazy val offsetTableConfiguration: OffsetTableConfiguration = new OffsetTableConfiguration(
+    configuration, readSideConfig
+  )
+  lazy val jdbcOffsetStore: JdbcOffsetStore = new JdbcOffsetStore(slickProvider, system, offsetTableConfiguration,
+    readSideConfig)(executionContext)
+
+  // FIXME what about OffsetStore? OffsetStore is internal. Guice module defines:
+  //       bind(classOf[OffsetStore]).to(Key.get(classOf[JdbcOffsetStore]))
+  //  private[lagom] lazy val offsetStore: OffsetStore = jdbcOffsetStore
```

Comment:
@jroper but then something is wrong, since `OffsetStore` is in the [internal](https://github.com/lagom/lagom/blob/1.2.x/persistence/src/main/scala/com/lightbend/lagom/internal/persistence/OffsetStore.scala) package